text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,212 @@
+an L1 construct for each one. This includes:
+
+1. Generating the construct class that extends `CfnResource`. I think we can use submodules to
+   represent the type namespace (`awsqs.eks` in this case), and the name of the resource as the construct name `Cluster`.
+2. Generate an `XxxProps` JSII struct for this construct based on the `Schema` field.
+   This can be done using [`json2jsii`](https://github.com/cdklabs/json2jsii), which is the same
+   tool we use to implemented the `import` command in CDK8s and CDKtf.
+
+### Is this a breaking change?
+
+Nope.
+
+### What are the drawbacks of this solution?
+
+> Describe any problems/risks that can be introduced if we implement this RFC.
+
+### What alternative solutions did you consider?
+
+#### `cdk import` versus library
+
+As mentioned above, an alternative approach would be to add support for `import`
+in the CDK CLI, which is inline with how other CDKs implement this functionality
+(`import` in CDK8s and `get` in CDKtf). The `import` experience offers a "just
+in time" option that the pre-published library option does not, but also
+increases the cognitive load for users since they will need to understand how
+`import` works.
+
+The downsides of the `import` approach (at this point) are that (a) it is a
+bigger investment as it involves a change to the CDK CLI; and (b) the Construct
+Hub won't be able to surface these types automatically (we will need some custom
+support for "imported" types, which is something that we would like to add to
+the Construct Hub in the future for all CDK domains).
+
+To make sure this path is possible in the future, the code generator will be
+implemented as a separate tool called `cdk-import`, which we can later integrate
+into the CDK CLI.
+
+#### Monolithic versus module per resource
+
+As mentioned above, due to the fact that resource schemas are versioned and we want to allow users to select which
+version of the resource they use, we determined that the best approach is to publish a generated module for each resource.
+
+### What is the high level implementation plan?
+
+* [ ] Implement [`cdk-import`](https://github.com/cdklabs/cdk-import) as a command-line tool that generates L1 constructs for registry extensions.
+* [ ] Create a mono-repo-style project using projen that utilizes `cdk-import` to generate a module for each registry resource.
+* [ ] Created a scheduled job which generates the modules periodically (should probably run against `us-east-1`)
+
+### Are there any open issues that need to be addressed later?
+
+* __Property name conversion__: `json2jsii` converts field names to camelCase to adhere with naming typescript naming conventions.
+  This means that we need to map field names back when we define the `CfnResource`. I think we might want to add a
+  feature to `json2jsii` that will generate conversion functions that can be used to convert back data types to the original schema.
```

Comment:
I think this can be removed since it has been resolved.

Reply:
Done. Thanks!